### ISSUE TITLE
Search results are not dynamically updated when new files are added to the library

### ISSUE THREAD

**Issue by peterhoeg** [strawberrymusicplayer/strawberry#1198]
- [X] I have checked the [FAQ](https://wiki.strawberrymusicplayer.org/wiki/FAQ) for answers.
- [X] I have checked the [Changelog](https://github.com/strawberrymusicplayer/strawberry/blob/master/Changelog) that the issue is not already fixed. 
- [X] I believe this issue is a bug, and not a general technical issue, question or feature requests that can be discussed on the [forum](https://forum.strawberrymusicplayer.org/). 

**Describe the bug**
If I do the following:

1. assume I have one album for the artist "Foobar"
2. I search in strawberry for "artist:foobar"
3. I then add a second album into one of the folders that strawberry searches for files
4. I click "Update changed collection folders"
5. Strawberry still only shows me the single album
6. If I then edit anything in my search, like removing the last "r" so that the search is "artist:fooba", the 2nd album now pops in

**To Reproduce**
Steps above should do the trick.

**Expected behavior**
Basically, I would like for new albums added to automatically update the active search (if any) so that strawberry  shows me both albums after it's done updating its database without having to touch the query string.

**System Information:**
 - Operating system: NixOS 23.05
 - Strawberry Version: 1.0.17 but this behaviour has been around for as long as I can remember
